### PR TITLE
Add new needRelocationsForCurrentMethodPC query

### PR DESCRIPTION
**Pull request by dchopra001**
Signed-off-by: Dhruv Chopra <Dhruv.C.Chopra@ibm.com>

### CODE REVIEW

**Comment by dchopra001** [comment]
See https://github.com/eclipse/omr/pull/4442 for related PR and motivation behind introducing this query.

@ymanton Can I get a review please?

File: runtime/compiler/env/VMJ9Server.hpp [hunk 1]
```diff
@@ -54,6 +54,7 @@ class TR_J9ServerVM: public TR_J9VM
    virtual bool needRelocationsForBodyInfoData() override                { return true; }
    virtual bool needRelocationsForPersistentInfoData() override          { return true; }
    virtual bool needRelocationsForLookupEvaluationData() override        { return true; }
+   virtual bool needRelocationsForStatics() override                     { return true; }
```
Comment (ymanton) [inline runtime/compiler/env/VMJ9Server.hpp]: you meant `needRelocationsForCurrentMethodPC` here I think?

Comment (dchopra001) [reply runtime/compiler/env/VMJ9Server.hpp]: Oops! Fixed in: b5c4f331a1

**Comment by ymanton** [comment]
Jenkins test sanity.functional all jdk8,jdk13 depends eclipse/omr#4442

**Comment by mpirvu** [comment]
Jenkins test sanity.functional all jdk8

**Comment by mpirvu** [comment]
On Java8 pLinux there are tap parsing errors
```
18:33:12  ]: special characters are not allowed
18:33:12  	at org.tap4j.parser.Tap13Parser.parseDiagnostics(Tap13Parser.java:469)
18:33:12  	at org.tap4j.parser.Tap13Parser.parseLine(Tap13Parser.java:288)
18:33:12  	at org.tap4j.parser.Tap13Parser.parseTapStream(Tap13Parser.java:252)
18:33:12  	... 14 more
18:33:12  Caused by: unacceptable character '?' (0xD852) special characters are not allowed
18:33:12  in "'string'", position 903
18:33:12  	at org.yaml.snakeyaml.reader.StreamReader.checkPrintable(StreamReader.java:67)
18:33:12  	at org.yaml.snakeyaml.reader.StreamReader.<init>(StreamReader.java:47)
18:33:12  	at org.yaml.snakeyaml.Yaml.load(Yaml.java:369)
18:33:12  	at org.tap4j.parser.Tap13Parser.parseDiagnostics(Tap13Parser.java:459)
18:33:12  	... 16 more
18:33:12  TAP parse errors found in the build. Marking build as UNSTABLE
18:33:12  TAP Reports Processing: FINISH
```
but looking at the console there is another error as well
```
18:33:12           [ERR] FAILURES DETECTED. Number of failed tests: 1
18:33:12           [ERR] 
18:33:12           [ERR] 1: j9dump_test_create_dump_from_signal_handler
18:33:12           [ERR] 	j9dumpTest.c line  311: j9dump_create returned: 1, with filename: The core file created by child process with pid = 13588 was not found. Expected to find core file with name "/home/jenkins/jenkins-agent/workspace/Test_openjdk8_j9_sanity.functional_ppc64le_linux_Personal/openjdk-tests/TKG/test_output_15759301498208/cmdLineTester_pltest_0/core"
18:33:12           [ERR] 		LastErrorNumber: -108
18:33:12           [ERR] 		LastErrorMessage: No such file or directory
```
This does not seem to be related to this PR. I'll trigger the test one more time to be sure.

**Comment by mpirvu** [comment]
Jenkins test sanity.functional plinux jdk8

**Comment by mpirvu** [comment]
`07:59:36  Failed in branch Pipeline_Build_Test_JDK8_x86-64_linux_jit`

Will trigger again

**Comment by mpirvu** [comment]
Yet another infra failure
```
07:48:05  Starting building: Pipeline_Build_Test_JDK8_ppc64le_linux #1326
Creating placeholder flownodes because failed loading originals.
Setting status of b5c4f331a12da5f699b1d1924ebad0ef2d9fffaa to FAILURE with url https://ci.eclipse.org/openj9/job/PullRequest-OpenJ9/2178/ and message: ' '
Using context: Pull Request - OpenJ9
java.io.IOException: Tried to load head FlowNodes for execution Owner[PullRequest-OpenJ9/2178:PullRequest-OpenJ9 #2178] but FlowNode was not found in storage for head id:FlowNodeId 1:93
	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution.initializeStorage(CpsFlowExecution.java:679)
	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution.onLoad(CpsFlowExecution.java:716)
	at org.jenkinsci.plugins.workflow.job.WorkflowRun.getExecution(WorkflowRun.java:665)
	at org.jenkinsci.plugins.workflow.job.WorkflowRun.onLoad(WorkflowRun.java:529)
```

**Comment by mpirvu** [comment]
Jenkins test sanity.functional plinux jdk8

**Comment by dchopra001** [comment]
@mpirvu All tests passed, so this is good to merge finally!

**Pull request merged**